### ISSUE TITLE
Question: Huffman tree impact on small game

### ISSUE THREAD

**Issue by ItSkary** [lichess-org/compression#16]
Greeting and my apologies for using the issue tracker for just a question but here what I would ask :

I just read Lichess post about this compressor
https://lichess.org/blog/Wqa7GiAAAOIpBLoY/developer-update-275-improved-game-compression

And I find it really brilliant, but then I started to remember courses I took lot of years ago and just curious about how such implementation perform against different chess game sizes.

I am using my memory about Huffman encoding, so probably I am making mistakes, but I remember that can be expensive to store the tree for short messages. And for that reason is not always used as compression algorithm. 
For that reason I am curious about data on this problem (compression of chess games) to understand if there is some threshold where simpler compression algorithm are cheaper (eg an adaptation of Zigzag VLQ because we are talking about byte/int).

Another way to cut head of small games, and improving in general compression rate may be also the use of shared opening book. So you can use just a number to pinpoint a sequence in the opening.

So to summarize, my question did you already thought about those two possible improvements? I think probably yes, then if possible can you please share your consideration about splitting the compression phase, because I am really curious about it.

**Comment by niklasf** [comment]
Hi.

(1) Compression here uses a Huffman tree that has been prepared ahead of time, based on a sample of games, for all games. So there's no overhead for storing the tree. It's hardcoded here:

https://github.com/lichess-org/compression/blob/e675db1936eaf6dc0c5f718366760f00a5b7b47d/src/main/java/game/Huffman.java#L64-L74

(2) We do not currently use an opening book. I think it's likely that doing so would be a small improvement.

**Comment by ItSkary** [comment]
Make a lot of sense, thanks for explanation.

In regards of point 2 I am still not persuaded, but I will not bother you while I dig deeper on that.

Thanks again for fast and clear response.
